roles/tests-integration/tests/common/proxy/mod.rs

```diff
@@ -0,0 +1,489 @@
+mod connection;
+mod error;
+
+use std::{net::SocketAddr, sync::Arc};
+
+use codec_sv2::{HandshakeRole, Initiator, Responder, StandardEitherFrame, StandardSv2Frame};
+use connection::Connection;
+use error::ProxyError;
+use key_utils::{Secp256k1PublicKey, Secp256k1SecretKey};
+use roles_logic_sv2::{
+    parsers::{
+        CommonMessages,
+        JobDeclaration::{
+            AllocateMiningJobToken, AllocateMiningJobTokenSuccess, DeclareMiningJob,
+            DeclareMiningJobError, DeclareMiningJobSuccess, IdentifyTransactions,
+            IdentifyTransactionsSuccess, ProvideMissingTransactions,
+            ProvideMissingTransactionsSuccess, SubmitSolution,
+        },
+        PoolMessages,
+        TemplateDistribution::{self, CoinbaseOutputDataSize},
+    },
+    utils::Mutex,
+};
+use tokio::{
+    net::{TcpListener, TcpStream},
+    select,
+    sync::mpsc::{channel, Receiver, Sender},
+};
+
+use codec_sv2::framing_sv2::framing::Frame as EitherFrame;
+
+type MessageFrame = StandardEitherFrame<PoolMessages<'static>>;
+
+#[derive(Debug, Clone)]
+pub struct SV2Proxy {
```

Comment:
I guess the original inspiration from `demand_easy_sv2` led you in this direction, but I think that using the term `Proxy` in this context can be problematic in the future, because it could very easily be confused with an actual SRI Proxy implementation that is being tested.

That applies to this `SV2Proxy` struct, the `ProxyBuilder` and the `proxy` module.

I would replace the term `Proxy` with something that implies that this is part of the integration test framework and not a role itself.

`Probe` is the first term that comes to mind, because in the context of instrumentation a probe is the tool that is used to collect data for analysis. That would avoid the possibility of confusing these objects with an actual role implementation.

Reply:
Yea good point.

Probe is a good option. Sniffer is another one.